packages/connect-web/src/connect-transport.ts

```diff
@@ -261,15 +263,39 @@ export function createConnectTransport(
 
       async function createRequestBody(
         input: AsyncIterable<I>
-      ): Promise<Uint8Array> {
-        if (method.kind != MethodKind.ServerStreaming) {
+      ): Promise<Uint8Array | ReadableStream> {
+        if (method.kind == MethodKind.ServerStreaming) {
+          const r = await input[Symbol.asyncIterator]().next();
+          if (r.done == true) {
+            throw "missing request message";
+          }
+          return encodeEnvelope(0, serialize(r.value));
+        } else if (
+          method.kind === MethodKind.ClientStreaming &&
+          supportsRequestStreams()
+        ) {
+          const inputStream = new ReadableStream({
+            async pull(controller) {
+              try {
+                const { value, done } = await input[
+                  Symbol.asyncIterator
+                ]().next();
```

Comment:
Calling `[Sybol.asyncIterator]()` on an iterable returns an iterator. But it might be a new one every time you call the method. The iterator should be obtained once for the `ReadableStream`. The stream should also call `throw` on the iterable. See the `iterableToReadableStream` function [here](https://github.com/bufbuild/connect-es/blob/6fdde58cca89415fe44c0f88a18881c4e79d0ef3/packages/connect/src/protocol/universal-fetch.ts#L112) for an example.